pkg/cmd/pulumi/plugin_install.go

```diff
@@ -174,6 +175,7 @@ func newPluginInstallCmd() *cobra.Command {
 					if err != nil {
 						return fmt.Errorf("%s downloading from %s: %w", label, install.PluginDownloadURL, err)
 					}
+					defer contract.IgnoreError(os.Remove(r.Name()))
```

Comment:
This will delete the file right away.
`defer` is evaluated until the outermost expression.
Surprised none of the tests caught this.